org.eclipse.gef/src/org/eclipse/gef/tools/DelayedDirectEditHelper.java

```diff
@@ -116,7 +117,10 @@ public void run() {
 			viewer.getControl().removeMouseListener(mouse);
 			viewer.getControl().removeKeyListener(key);
 		}
-		activeHelper = null;
+		setActiveHelper(null);
 	}
 
+	public static synchronized void setActiveHelper(DelayedDirectEditHelper activeHelper) {
```

Comment:
Should this method be public?
```suggestion
	private static synchronized void setActiveHelper(DelayedDirectEditHelper activeHelper) {
```

Reply:
no it should be private. Thx for your sharp eyes.